src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1371,6 +1389,11 @@ static OptoReg::Name find_first_set(LRG &lrg, RegMask mask, int chunk) {
         }
       }
       return OptoReg::Bad; // will cause chunk change, and retry next chunk
+    } else if (lrg._is_predicate) {
```

Comment:
lrg._is_predicate is set to 1 for all architectures, so using scalable_reg_slots on this path doesn't look correct.

Reply:
This is inside lrg.is_scalable() condition of L1362, which handles scalable predicate register specifically. I think for non-scalable arch, the normal path of regmask handling should be fine.

Reply:
I missed that. Thanks, this looks good.